grid2op/tests/test_alert_trust_score.py

```diff
@@ -0,0 +1,1767 @@
+    ts_attack = np.array(kwargs_opponent["steps_attack"])
+    res = []
+    for i, ts in enumerate(ts_attack):
+        if not multi:
+            res.append(ts + np.arange(kwargs_opponent["duration"]))
+        else:
+            res.append(ts + np.arange(kwargs_opponent["duration"][i]))
+    return np.unique(np.concatenate(res).flatten())
+
+    
+class TestOpponent(GeometricOpponent): 
+    """An opponent that can select the line attack, the time and duration of the attack."""
+    
+    def __init__(self, action_space):
+        super().__init__(action_space)
+        self.custom_attack = None
+        self.duration = None
+        self.steps_attack = None
+
+    def init(self, partial_env,  lines_attacked=[ATTACKED_LINE], duration=10, steps_attack=[0,1]):
+        super().init(partial_env, lines_attacked, )
+        attacked_line = lines_attacked[0]
+        self.custom_attack = self.action_space({"set_line_status" : [(l, -1) for l in lines_attacked]})
+        self.duration = duration
+        self.steps_attack = steps_attack
+        self.env = partial_env
+
+    def attack(self, observation, agent_action, env_action, budget, previous_fails): 
+        if observation is None:
+            return None, None
+        current_step = self.env.nb_time_step
+        if current_step not in self.steps_attack: 
+            return None, None
+        
+        return self.custom_attack, self.duration
+
+
+class TestOpponentMultiLines(GeometricOpponentMultiArea): 
+    """An opponent that can select the line attack, the time and duration of the attack."""
+    
+    def __init__(self, action_space):
+        super().__init__(action_space)
+        self.custom_attack = None
+        self.duration = None
+        self.steps_attack = None
+
+    def init(self, partial_env,  lines_attacked=[ATTACKED_LINE], duration=[10,10], steps_attack=[0,1]):
+        super().init(partial_env, [[l] for l in lines_attacked])
+        attacked_line = lines_attacked[0]
+        self.custom_attack = [ self.action_space({"set_line_status" : [(l, -1)]}) for l in lines_attacked]
+        self.duration = duration
+        self.steps_attack = steps_attack
+        self.env = partial_env
+        
+    def attack(self, observation, agent_action, env_action, budget, previous_fails): 
+        if observation is None:
+            return None, None
+
+        current_step = self.env.nb_time_step
+        if current_step not in self.steps_attack: 
+            return None, None
+        
+        index = self.steps_attack.index(current_step)
+
+        return self.custom_attack[index], self.duration[index]
+
+
+# Test alert blackout / tets alert no blackout
+class TestAlertTrustScoreNoBlackout(unittest.TestCase):
+    """test the basic bahavior of the assistant alert feature when no attack occur """
+
+    def setUp(self) -> None:
+        self.env_nm = os.path.join(
+            PATH_DATA_TEST, "l2rpn_idf_2023_with_alert"
```

Comment:
why not using the "l2rpn_idf_2023" environment ?

Reply:
there is no l2rpn_idf_2023 in "data_test". Isn't it from where you load grid2op environments when doing tests ?
The l2rpn_idf_2023 in "data" has long scenarios. In case we use that one we should probably reduce the max_iter to the number of steps run in l2rpn_idf_2023_with_alert

Reply:
I suggest we keep the l2rpn_idf_2023_with_alert env for now in these tests as originally done in these tests when I took over and as far of knowledge I don't see better option so far. Could make a change of env later if relevant